### PR TITLE
CLI: Resolve the enrollment url from DID

### PR DESCRIPTION
**Pull request by algattik**
## What this PR changes/adds

Add an option to Registration Service CLI to provide a dataspace did instead of Registration Service API url directly.
The Registration Service CLI resolves the did document and fetches the enrollment url from it.

To keep the backwards compatibility the input option for providing the service url directly is not removed.
It will be removed once the MVD will use the CLI with did as an input.

## Why it does that

The enrollment url shouldn't be used directly but it should be a part of a Dataspace did document.

## Further notes

To resolve the did document the CLI uses the WebDidResolver class from EDC. 

## Linked Issue(s)

linked to https://github.com/agera-edc/MinimumViableDataspace/issues/43

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [ ] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/RegistrationService/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/RegistrationService/blob/main/styleguide.md) for details_)

### CODE REVIEW

File: client-cli/src/main/java/org/eclipse/dataspaceconnector/registration/cli/RegistrationUrlResolver.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+     * Resolves the DID document from did:web.
+     */
+    private final DidResolver resolver;
+
+    public RegistrationUrlResolver(DidResolver resolver) {
+        this.resolver = resolver;
+    }
+
+    @NotNull
+    public Optional<String> resolveUrl(String did) {
```
Comment (paullatzelsperger) [inline client-cli/src/main/java/org/eclipse/dataspaceconnector/registration/cli/RegistrationUrlResolver.java:42]: Not a big fan of using `Optional` as method return value. I'd prefer returning a `@Nullable` String or, even better, a `Result<String>`

**Review by paullatzelsperger — COMMENTED** [review]
A few remarks:
- a test for the `AddParticipantCommand` is missing.
- the `VerifiableCredentialsFactory` is not available anymore in latest EDC builds, it was renamed to `JwtUtils` and made package-private
- please don't use `Optional` in methods signatures or field declarations (see inline comment). We have established that as a principle in EDC and want to have consistency across repos.

**Comment by algattik** [comment]
> A few remarks:
> 
> * a test for the `AddParticipantCommand` is missing.

There is `org.eclipse.dataspaceconnector.registration.cli.ParticipantsCommandTest#add`

> * the `VerifiableCredentialsFactory` is not available anymore in latest EDC builds, it was renamed to `JwtUtils` and made package-private

Where do you see `VerifiableCredentialsFactory` being used?

> * please don't use `Optional` in methods signatures or field declarations (see inline comment). We have established that as a principle in EDC and want to have consistency across repos.

Refactored to `Result`.

**Comment by paullatzelsperger** [comment]
> Where do you see VerifiableCredentialsFactory being used?

It is in `ClientUtils` and `ClientUtilsTest`

**Comment by algattik** [comment]
> > Where do you see VerifiableCredentialsFactory being used?
> 
> It is in `ClientUtils` and `ClientUtilsTest`

That is not in scope of this PR. We can create a separate issue to refactor if needed.
Note that the scope of MVD is EDC milestone 5, hence we have it available for now.

**Review by paullatzelsperger — APPROVED** [review]

**Pull request merged**